lib/websocket_lwt.ml

```diff
@@ -109,18 +109,33 @@ let set_tcp_nodelay flow =
   | TCP { fd; _ } -> Lwt_unix.setsockopt fd Lwt_unix.TCP_NODELAY true
   | _ -> ()
 
+module SSet = Set.Make(String)
+
+let check_origin ?(origin_mandatory=false) ~hosts =
```

Comment:
I think `hosts` should be a list, not a set:

- You're using `.exists`, so any speed advantages are voided
- The number of hosts is usually small